cmd/lakectl/cmd/local_init.go

```diff
@@ -0,0 +1,83 @@
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/spf13/cobra"
+	"github.com/treeverse/lakefs/pkg/git"
+	"github.com/treeverse/lakefs/pkg/uri"
+)
+
+const (
+	initMinArgs = 1
+	initMaxArgs = 2
+)
+
+var localInitCmd = &cobra.Command{
+	Use:   "init [directory] <lakeFS path URI>",
+	Short: "set a local directory to sync with a lakeFS ref and path",
+	Args:  cobra.RangeArgs(initMinArgs, initMaxArgs),
+	Run: func(cmd *cobra.Command, args []string) {
+		var remote *uri.URI
+		dir := "."
+		if len(args) == initMaxArgs {
+			dir = args[0]
+			remote = MustParsePathURI("remote ref", args[1])
+		} else {
+			remote = MustParsePathURI("remote ref", args[0])
+		}
+
+		localPath, err := filepath.Abs(dir)
+		if err != nil {
+			DieErr(err)
+		}
+
+		stat, err := os.Stat(localPath)
+		switch {
+		case errors.Is(err, os.ErrNotExist):
+			if confirmation, err := Confirm(cmd.Flags(), fmt.Sprintf("Directory '%s' doesn't exist, create", localPath)); err != nil || !confirmation {
+				Die("local init aborted", 1)
+			}
+			if err := os.MkdirAll(dir, os.ModePerm); err != nil {
+				DieFmt("Failed to create dir: %s", localPath)
+			}
```

Comment:
Suggest not to be interactive and just use a flag to identify if we like to create directory if not exists. Alternative just let the use create the target path - less code less issues.